src/core/CoolBoard.cpp

```diff
@@ -377,11 +379,49 @@ void CoolBoard::readBoardData(JsonObject &reported) {
   }
 }
 
-void CoolBoard::sleep(unsigned long interval) {
-  if (interval > 0) {
-    INFO_VAR("Going to sleep for seconds:", interval);
-    ESP.deepSleep((uint64(interval) * 1000000ULL), WAKE_RF_DEFAULT);
+void CoolBoard::sleep() {
+  EEPROM.begin(5);
+  delay(5000);
+  uint8_t val[4];
+  rst_info *resetInfo; 
+  resetInfo = ESP.getResetInfoPtr();
+  if (!EEPROM.read(0x04) ||
```

Comment:
No, you should stop checking for `BOOTSTRAP_PIN` and only continue to `deepSleep` if the reset reason is `DEEP_SLEEP_AWAKE`. Here, your check has no effect. It should be something like
```cpp
if (!EEPROM.read(0x04) || resetInfo->reason != REASON_DEEP_SLEEP_AWAKE) {
  // reset the counter
}
```